### PR TITLE
Fips verification

### PR DESCRIPTION
**Pull request by jimdevops19**
New check for FIPS.
This merge will allow checking if FIPS is installed correctly in the entire openshift-storage namespace.
Runs after the installation of OCS. (within the ocs_install_verification())

### CODE REVIEW

**Review by liranmauda — APPROVED** [review]
LGTM

File: ocs_ci/ocs/resources/pod.py [hunk 1]
```diff
@@ -1455,3 +1455,20 @@ def check_pods_in_running_state(namespace=defaults.ROOK_CLUSTER_NAMESPACE):
...
+    checks the running state pods in a given namespace.
+    Returns:
+        List: all the pod objects that are in running state only
+    """
+    list_of_pods = get_all_pods(namespace)
+    ocp_pod_obj = OCP(kind=constants.POD, namespace=namespace)
+    running_pods_object = list()
+    for p in list_of_pods:
+        status = ocp_pod_obj.get_resource(p.name, 'STATUS')
+        if status in "Running":
```
[the remark this answers is not in the document]
Comment (jimdevops19) [reply ocs_ci/ocs/resources/pod.py]: > shouldn't it be `if "Running" in status:`?
Yes:)

File: ocs_ci/ocs/resources/fips.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    """
+    Checks if FIPS is activated on all pods
+
+        Args:
+            fips_location: File that refers to fips, written 1 if enabled,
+                0 otherwise
+        Raises:
+            FIPS not installed Exception.
+                Value of fips location file should include 1
+                    in all pods within the given namespace
```
Comment (petr-balogh) [inline ocs_ci/ocs/resources/fips.py:19]:
```suggestion
            FipsNotInstalledException: If the value of fips location file does not include 1
                in all pods within the given namespace.
```

File: ocs_ci/ocs/resources/fips.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    """
+
+    running_pods_object = pod.get_running_state_pods()
+    for running_pod in running_pods_object:
+        fips_value = running_pod.exec_sh_cmd_on_pod(f"cat {fips_location}")
+        if str(fips_value).strip() != "1":
+            raise FipsNotInstalledException(
+                "Error in the installation of FIPS on the cluster!"
+                f"Found value different than 1 in pod {running_pod.name}"
+                f"Value: \n {fips_value}"
```
Comment (petr-balogh) [inline ocs_ci/ocs/resources/fips.py:30]: Do we need new line here? `\n` ?  I think that better will be just to see this kind of log message:  `Value: 0` for example

File: ocs_ci/ocs/resources/pod.py [hunk 4]
```diff
@@ -1455,3 +1455,22 @@ def check_pods_in_running_state(namespace=defaults.ROOK_CLUSTER_NAMESPACE):
...
+    Checks the running state pods in a given namespace.
+
+        Returns:
+            List: all the pod objects that are in running state only
+
+    """
+    list_of_pods = get_all_pods(namespace)
+    ocp_pod_obj = OCP(kind=constants.POD, namespace=namespace)
+    running_pods_object = list()
+    for p in list_of_pods:
```
Comment (petr-balogh) [inline ocs_ci/ocs/resources/pod.py:1470]: Avoid of using just one letter variable name. Rather use whole `pod` variable name instead of just `p`.

**Review by petr-balogh — COMMENTED** [review]
You haven't signed all the commits. You can squash them into one now and sign with `git commit --amend -s`

File: ocs_ci/ocs/resources/fips.py (hunk 3 above)
Comment (jimdevops19) [reply ocs_ci/ocs/resources/fips.py:30]: Agree,  `Value:  0 ` looks nicer

File: conf/ocp_version/ocp-4.5-ga-minus1-config.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+---
```
Comment (ebenahar) [inline conf/ocp_version/ocp-4.5-ga-minus1-config.yaml:1]: Do these changes suppose to be part of this PR?

File: ocs_ci/deployment/flexy.py [hunk 6]
```diff
@@ -88,8 +88,28 @@ def deploy_prereq(self):
...
             config.FLEXY['VARIABLES_LOCATION'] = self.template_file
         config.FLEXY['INSTANCE_NAME_PREFIX'] = self.cluster_name
+        config.FLEXY['LAUNCHER_VARS'].update(
+            self.get_installer_payload()
+        )
         self.merge_flexy_env()
 
+    def get_installer_payload(self, version=None):
+        """
+        A proper installer payload url required for flexy
```
Comment (ebenahar) [inline ocs_ci/deployment/flexy.py:98]: Please add `Args` and `Returns` to the docstring

File: ocs_ci/ocs/ocs_upgrade.py [hunk 7]
```diff
@@ -119,10 +119,14 @@ def verify_image_versions(old_images, upgrade_version):
...
-            old_images, selector=constants.RGW_APP_LABEL, count=rgw_count
-        )
+        # Workaround for https://bugzilla.redhat.com/show_bug.cgi?id=1857802 - RGW count is 1
+        # post upgrade to OCS 4.5. Tracked with
+        # https://github.com/red-hat-storage/ocs-ci/issues/2532
+        # TODO: uncomment the below 1 line:
+        # rgw_count = 2 if float(config.ENV_DATA['ocs_version']) >= 4.5 else 1
+        # TODO: Delete the below 1 line
+        rgw_count = 1
+        verify_pods_upgraded(old_images, selector=constants.RGW_APP_LABEL, count=rgw_count)
```
Comment (ebenahar) [inline ocs_ci/ocs/ocs_upgrade.py:129]: I think this commit includes rebase code

File: conf/ocp_version/ocp-4.5-ga-minus1-config.yaml (hunk 5 above)
Comment (jimdevops19) [reply conf/ocp_version/ocp-4.5-ga-minus1-config.yaml]: No, fixed the rebase into squashing them

**Review by ebenahar — APPROVED** [review]

**Review by fbalak — APPROVED** [review]

**Review by petr-balogh — APPROVED** [review]
LGTM

**Pull request merged**